crates/sui-bridge/src/eth_syncer.rs

```diff
@@ -156,16 +156,23 @@ where
                 continue;
             };
             let len = events.len();
-            if !events.is_empty() {
-                // Note: it's extremely critical to make sure the Logs we send via this channel
-                // are complete per block height. Namely, we should never send a partial list
-                // of events for a block. Otherwise, we may end up missing events.
-                events_sender
-                    .send((contract_address, events))
-                    .await
-                    .expect("All Eth event channel receivers are closed");
-                tracing::info!(?contract_address, "Observed {len} new Eth events",);
-            }
+
+            // Note 1: we send an empty list of events to the channel. This is because of how
+            // `eth_getLogs` api is designed - we want cursor to move forward continuously.
+
+            // Note 2: it's extremely critical to make sure the Logs we send via this channel
+            // are complete per block height. Namely, we should never send a partial list
+            // of events for a block. Otherwise, we may end up missing events.
+            events_sender
+                .send((contract_address, end_block, events))
+                .await
+                .expect("All Eth event channel receivers are closed");
+            tracing::info!(
+                ?contract_address,
+                start_block,
+                end_block,
+                "Observed {len} new Eth events",
+            );
```

Comment:
meat 2 of this PR - Eth syncer's sends event list to orchestrator even when the event list is empty